core/src/main/java/net/akami/yggdrasil/player/YggdrasilPlayer.java

```diff
@@ -1,126 +1,148 @@
+        this.id = id;
+        this.spellHandler = spellHandler;
+        this.sequence = new ArrayList<>();
+        this.mana = new PlayerManaContainer(100, 4f, this);
+        this.life = new PlayerLifeComponent(3, 50, this);
+        this.excludedTypes = new ArrayList<>();
+        this.spells = new ArrayList<>();
+        this.items = new ArrayList<>();
+        this.actionBarSpellDisplayer = actionBarSpellDisplayer;
+        addDefaultItems();
+        addDefaultSpells();
+    }
+
+    @Override
+    public void addDefaultItems() {
+        items.addAll(Arrays.asList(
+                new AdvancedMovementItem(this),
+                new InstantHealItem(this),
+
+                new SpellTriggerItem(this, this),
+                new FireElementItem(this, this),
+                new AirElementItem(this, this),
+                new EarthElementItem(this, this),
+                new WaterElementItem(this, this)));
+        addItemsToInventory();
+    }
+
+    private void addItemsToInventory() {
+        Optional<Player> optPlayer = Sponge.getServer().getPlayer(id);
+        optPlayer.ifPresent(this::fill);
+    }
+
+    private void fill(Player target) {
+        target.getInventory().clear();
+        for(InteractiveItem item : items) {
+            ItemUtils.fitItemInInventory(target, item);
+        }
+    }
+
+    @Override
+    public void addDefaultSpells() {
+        spells.addAll(Arrays.asList(
+                new FireballCaster(this),
+                new PhoenixArrowCaster(),
+                new IncendiaCaster(),
+                new CounterVelocityCaster(this),
+                new WaterPrisonCaster()
+        ));
+    }
+
+    @Override
+    public UUID getUUID() {
+        return id;
+    }
+
+    @Override
+    public LifeComponent getLife() {
+        return life;
+    }
+
+    @Override
+    public ManaContainer getMana() {
+        return mana;
+    }
+
+    @Override
+    public List<SpellCaster> getSpells() {
+        return spells;
+    }
+
+    @Override
+    public List<InteractiveItem> getItems() {
+        return items;
+    }
+
+    @Override
+    public List<ElementType> currentSequence() {
+        return sequence;
+    }
+
+    @Override
+    public List<SpellType> currentlyExcludedTypes() {
+        return excludedTypes;
+    }
+
+    @Override
+    public ExcludedSpellHandler getExclusionHandler() {
+        return spellHandler;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        return obj instanceof YggdrasilPlayer && id.equals(((YggdrasilPlayer) obj).id);
+    }
+
+    @Override
+    public void displayActionBar(Text text) {
+        Sponge.getServer().getPlayer(this.id).ifPresent(player -> player.sendMessage(ChatTypes.ACTION_BAR, text));
+    }
+
+    @Override
+    public void addActionBarDisplayElement(Text element) {
+        actionBarSpellDisplayer.addElement(this, element);
```

Comment:
I believe that method is unnecessary, creating back and forth dependencies for no real reason, since the implementation won't change

Reply:
I've tried to find a solution for this, and it looks as though that method wasn't a bad idea after all, since that particular method is not called within the manager class. Therefore, the manager object is not necessarily present. <br>
However, since a manager object is always needed, maybe a getDisplayUpdater() method would be appropriate within the interface ?